test/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProjectWithGeneratePackageOnBuild.cs

```diff
@@ -71,7 +71,7 @@ public void It_builds_and_result_contains_dependencies_dll()
             }
         }
 
-        [Theory(Skip = "https://github.com/dotnet/sdk/issues/3471")]
+        [Theory]
```

Comment:
The 2 test cases with `generatePackageOnBuild: true` fail because of reasons similar to:
```
C:\Code\sdk\artifacts\bin\redist\Debug\dotnet\sdk\9.0.100-ci\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(221,5): error NU5026: The file 'C:\Code\sdk\artifacts\tmp\Debug\It_packs_succ---74529E27\bin\Debug\net9.0\HelloWorld.runtimeconfig.json' to be packed was not found on disk. [C:\Code\sdk\artifacts\tmp\Debug\It_packs_succ---74529E27\HelloWorld.csproj]
```

Seems like the test project it builds isn't building properly, and it goes to pack the project, but fails. The original issue was actually marked as a dup of another issue, so I'll put that issue as the Skip reason instead.